posthog/models/messaging.py

```diff
@@ -21,13 +21,14 @@ def get_or_create(self, defaults=None, **kwargs):
 
 
 class MessagingRecord(UUIDModel):
-
     objects = MessagingRecordManager()
 
     email_hash: models.CharField = models.CharField(max_length=1024)
     campaign_key: models.CharField = models.CharField(max_length=128)
     sent_at: models.DateTimeField = models.DateTimeField(null=True)
     created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
+    resend_frequency_days: models.IntegerField = models.IntegerField(null=True, blank=True)
+    resend_dates: models.JSONField = models.JSONField(default=list, null=True, blank=True)
 
     class Meta:
-        unique_together = ("email_hash", "campaign_key")  # can only send campaign once to each email
+        unique_together = ("email_hash", "campaign_key", "resend_frequency_days")  # one record per email per campaign
```

Comment:
random drop by: Would you send the same campaign to the same person multiple times based on different resend frequencies? (i.e say a 5 day resend, and a 10 day one, which means the person gets one on day 5, then 2 on day 10, and so on)

Or given a person and a campaign, there's a unique resend frequency set for them.

Zero context, but it seems I'd want the latter, in which case you can get rid of the extra constraint

Reply:
I considered that. My thinking was that if there is ever a change in resend frequency that there would be a new record created, otherwise it would be harder to debug ("this email should have sent every day, why did it only send after 7 days?")

But I do need the constraint so I will probably update to your suggestion.